storm-core/src/jvm/org/apache/storm/trident/TridentTopology.java

```diff
@@ -394,11 +395,28 @@ public StormTopology build() {
         Map<Node, String> spoutIds = genSpoutIds(spoutNodes);
         Map<Group, String> boltIds = genBoltIds(mergedGroups);
         
+        Map defaults = Utils.readDefaultConfig();
+
         for(SpoutNode sn: spoutNodes) {
             Integer parallelism = parallelisms.get(grouper.nodeGroup(sn));
+
+            Map<String, Number> spoutRes = null;
+            if(sn instanceof ITridentResource) {
+                spoutRes = mergeDefaultResources(((ITridentResource)sn).getResources(), defaults);
+            }
+            else {
+                spoutRes = mergeDefaultResources(null, defaults);
+            }
```

Comment:
Good catch. I think we can ignore the null check even, though. We don't insert nulls into the spoutNodes collection, and if there were some in there somehow, lots of stuff would blow up.